openvdb/openvdb/points/PointRasterizeFrustum.h

```diff
@@ -0,0 +1,1805 @@
+            } else {
+                ValueT weightValue = castValue<ValueT>(scale);
+                ValueT newValue = MultiplyOp<ValueT>::mul(weightValue, attributeScale);
+                if (scaleByVoxelVolume) {
+                    newValue /= static_cast<ValueT>(transform.voxelSize(ijk.asVec3d()).product());
+                }
+                if (point_rasterize_internal::greaterThan(newValue, threshold)) {
+                    if (isTemp) {
+                        tempValueAccessor->modifyValue(ijk, MaxOpT(newValue));
+                        if (tempWeightAccessor) {
+                            tempWeightAccessor->modifyValue(ijk, MaxOpT(weightValue));
+                        }
+                    } else {
+                        if (mComputeMax && !forceSum) {
+                            valueAccessor.modifyValue(ijk, MaxOpT(newValue));
+                        } else {
+                            valueAccessor.modifyValue(ijk, SumOpT(newValue));
+                        }
+                        if (weightAccessor) {
+                            weightAccessor->modifyValue(ijk, SumOpT(weightValue));
+                        }
+                    }
+                }
+            }
+        };
+
+        auto modifyVoxelOp = [&](const Coord& ijk, const double scale, const AttributeT& attributeScale)
+        {
+            doModifyVoxelOp(ijk, scale, attributeScale, /*isTemp=*/false, /*forceSum=*/false);
+        };
+
+        auto sumVoxelOp = [&](const Coord& ijk, const double scale, const AttributeT& attributeScale)
+        {
+            doModifyVoxelOp(ijk, scale, attributeScale, /*isTemp=*/false, /*forceSum=*/true);
+        };
+
+        auto doModifyVoxelByDistanceOp = [&](const Coord& ijk, const double scale, const AttributeT& attributeScale,
+            const double distanceSqr, const double radiusSqr, const bool isTemp)
+        {
+            if (distanceSqr >= radiusSqr)   return;
+            if (isBool) {
+                valueAccessor.modifyValue(ijk, TrueOp(scale));
+            } else {
+                double distance = std::sqrt(distanceSqr);
+                double radius = std::sqrt(radiusSqr);
+                double result = 1.0 - distance/radius;
+                doModifyVoxelOp(ijk, result * scale, attributeScale, isTemp, /*forceSum=*/false);
+            }
+        };
+
+        auto modifyVoxelByDistanceOp = [&](const Coord& ijk, const double scale, const AttributeT& attributeScale,
+            const double distanceSqr, const double radiusSqr)
+        {
+            doModifyVoxelByDistanceOp(ijk, scale, attributeScale, distanceSqr, radiusSqr, /*isTemp=*/false);
+        };
+
+        auto modifyTempVoxelByDistanceOp = [&](const Coord& ijk, const double scale, const AttributeT& attributeScale,
+            const double distanceSqr, const double radiusSqr)
+        {
+            doModifyVoxelByDistanceOp(ijk, scale, attributeScale, distanceSqr, radiusSqr, /*isTemp=*/true);
+        };
```

Comment:
Quite a lot of lambdas here which aren't used till later - could they get some one line docs?